src/components/FeedPostsList/FeedPostsList.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useState, useEffect, MouseEvent } from 'react';
+import Grid from '@mui/material/Grid';
+import Pagination from '@mui/material/Pagination';
+import useMediaQuery from '@mui/material/useMediaQuery';
+
+import FeedPost from './FeedPost';
+import {
+  MAX_POSTS,
+  POSTS_PER_PAGE,
+  MEDIA_QUERY,
+  ErrorMessages,
+} from '../../constants';
+import { TrendingFeedList, FeedPostItem } from '../../types/trendingFeedTypes';
+import { StyledGridContainer, StyledDiv } from './styles';
+import ErrorAlert from '../ErrorAlert';
+
+type Props = {
+  data: TrendingFeedList
+}
+
+const FeedPostsList = ({ data }: Props) => {
```

Comment:
Maybe better to rename prop 'data' to 'posts', so in this case, it will be easier to understand, what data is provided to the component. 
As anyway you will need to watch an additional file with type definitions, to see what exactly TrendingFeedList consists of.